pyspecdata/general_functions.py

```diff
@@ -128,38 +128,27 @@ def check_ascending_axis(u,tolerance = 1e-7,additional_message = []):
     assert du > 0, thismsg
     return du
 
-def init_logging(levelString, filename='pyspecdata.log'):
-    "Initialize logging on pyspecdata.log -- do NOT log if run from within a notebook (it's fair to assume that you will run first before embedding)"
-    root = logging.getLogger()
+def init_logging(level=logging.INFO, filename='pyspecdata.log'):
+    r"""Initialize logging on pyspecdata.log -- do NOT log if run from within a
+    notebook (it's fair to assume that you will run first before embedding)"""
+    if type(level) is str:
+        if level.lower() == 'info':
+            level=logging.INFO
+        elif level.lower() == 'debug':
+            level=logging.DEBUG
+        else:
+            raise ValueError("if you give me level as a string, give me 'info' or 'debug'")
```

Comment:
the strategy is to interpret "level" here, and then just use it below.  A few less lines than what you ended up with, but does the same thing -- also what's below is less complicated

Reply:
I was running into an error where the obj had no attribute "lower" and upon looking into this lower I could not find much. Could you maybe help explain this?

Reply:
I THINK this section is saying that if the lowest level reads 'info' (as in logger.info()) then the level will be that of INFO or 20 (not quite sure on the numerical conception but working on it). If there is no string to state the lowest level then it won't declare a level

Reply:
actually, you'll notice that in my edits, unrelated to what you were doing, I made the code more robust by checking for the case where `level` is set to a string.  `lower` is a method of the string class -- it converts everything to lowercase: This way, setting `level` to `"DEBUG"`, `"Debug"`, and `"debug"` has the same result

Reply:
ah okay this makes sense because I was actually wondering what the difference was between all caps and all lower case. But now my question is INFO and DEBUG are integers correct? so how can an integer have a method of lower?

Reply:
`logger.INFO` and `logger.DEBUG` are.  But the `level` argument passed to this function is not (this is just a convenience function, and that's part of the point -- to be able to avoid importing logger yourself and also having to type out `logger.INFO`

Reply:
gotcha